pkg/client/cache/eventqueue.go

```diff
@@ -324,11 +339,22 @@ func (eq *EventQueue) Replace(objects []interface{}, resourceVersion string) err
 	}
 
 	if len(eq.queue) > 0 {
+		eq.lastReplaceKey = eq.queue[len(eq.queue)-1]
 		eq.cond.Broadcast()
+	} else {
+		eq.lastReplaceKey = ""
 	}
 	return nil
 }
 
+// TODO doc
+func (eq *EventQueue) LastReplaceRead() bool {
```

Comment:
Does anyone actually care about this value or do they just want to know if they have `ListEventsPending`?

Also, the way this behaves ends up be kind of weird since I'll look to see if I've popped everything I listed, I'll get back "yes" and then later it will become false again on the re list.